drivers/display/display_ili9341.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ * Copyright (c) 2021 Krivorot Oleg <krivorot.oleg@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_DRIVERS_DISPLAY_DISPLAY_ILI9341_H_
+#define ZEPHYR_DRIVERS_DISPLAY_DISPLAY_ILI9341_H_
+
+#include <device.h>
+
+/* Commands/registers. */
+#define ILI9341_SWRESET 0x01
+#define ILI9341_GAMSET 0x26
+#define ILI9341_FRMCTR1 0xB1
+#define ILI9341_DISCTRL 0xB6
+#define ILI9341_ETMOD 0xB7
+#define ILI9341_PWCTRL1 0xC0
+#define ILI9341_PWCTRL2 0xC1
+#define ILI9341_VMCTRL1 0xC5
+#define ILI9341_VMCTRL2 0xC7
+#define ILI9341_PWCTRLA 0xCB
+#define ILI9341_PWCTRLB 0xCF
+#define ILI9341_PGAMCTRL 0xE0
+#define ILI9341_NGAMCTRL 0xE1
+#define ILI9341_TIMCTRLA 0xE8
+#define ILI9341_TIMCTRLB 0xEA
+#define ILI9341_PWSEQCTRL 0xED
+#define ILI9341_EN3G 0xF2
+#define ILI9341_PUMPRATIOCTRL 0xF7
+
+/* Commands/registers length. */
+#define ILI9341_GAMSET_LEN 1U
+#define ILI9341_FRMCTR1_LEN 2U
+#define ILI9341_DISCTRL_LEN 3U
+#define ILI9341_PWCTRL1_LEN 1U
+#define ILI9341_PWCTRL2_LEN 1U
+#define ILI9341_VMCTRL1_LEN 2U
+#define ILI9341_VMCTRL2_LEN 1U
+#define ILI9341_PGAMCTRL_LEN 15U
+#define ILI9341_NGAMCTRL_LEN 15U
+
+#define ILI9341_PWCTRLA_LEN 5U
+#define ILI9341_PWCTRLB_LEN 3U
+#define ILI9341_PWSEQCTRL_LEN 4U
+#define ILI9341_TIMCTRLA_LEN 3U
+#define ILI9341_TIMCTRLB_LEN 2U
+#define ILI9341_PUMPRATIOCTRL_LEN 1U
+#define ILI9341_EN3G_LEN 1U
+#define ILI9341_ETMOD_LEN 1U
+
+/** X resolution (pixels). */
+#define ILI9341_X_RES 240U
+/** Y resolution (pixels). */
+#define ILI9341_Y_RES 320U
+
+/** ILI9341 registers to be initialized. */
+struct ili9341_regs {
+	uint8_t gamset[ILI9341_GAMSET_LEN];
+	uint8_t frmctr1[ILI9341_FRMCTR1_LEN];
+	uint8_t disctrl[ILI9341_DISCTRL_LEN];
+	uint8_t pwctrl1[ILI9341_PWCTRL1_LEN];
+	uint8_t pwctrl2[ILI9341_PWCTRL2_LEN];
+	uint8_t vmctrl1[ILI9341_VMCTRL1_LEN];
+	uint8_t vmctrl2[ILI9341_VMCTRL2_LEN];
+	uint8_t pgamctrl[ILI9341_PGAMCTRL_LEN];
+	uint8_t ngamctrl[ILI9341_NGAMCTRL_LEN];
+
+	uint8_t pwctrla[ILI9341_PWCTRLA_LEN];
+	uint8_t pwctrlb[ILI9341_PWCTRLB_LEN];
+	uint8_t pwseqctrl[ILI9341_PWSEQCTRL_LEN];
+	uint8_t timctrla[ILI9341_TIMCTRLA_LEN];
+	uint8_t timctrlb[ILI9341_TIMCTRLB_LEN];
+	uint8_t pumpratio[ILI9341_PUMPRATIOCTRL_LEN];
+	uint8_t en3g[ILI9341_EN3G_LEN];
+	uint8_t entrymdset[ILI9341_ETMOD_LEN];
+};
+
+/* Initializer macro for ILI9341 registers. */
+#define ILI9341_REGS_INIT(n)                                                                       \
```

Comment:
Can you add build time asserts that each register set has the correct number of bytes specified?

The concern is that if a user specifies too few bytes.  The calls to ili9xxx_transmit() use a fixed byte 
length, so ili9xxx_transmit() will write garbage data.

This could be added as part of the ILI9341_REGS_INIT() macro, inserted before the ili9xxx_regs_##n definition:
BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9341), gamset), ILI9341_GAMSET_LEN) \

Reply:
Do I need to write a comment to ASSERT?
If so, then this is enough:

`BUILD_ASSERT(DT_PROP_LEN(DT_INST(n, ilitek_ili9341), gamset) != ILI9341_GAMSET_LEN,        \
		     "ili9341: Error length gamma set (GAMSET) register");                         \`